src/components/TableHead.tsx

```diff
@@ -0,0 +1,57 @@
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { SearchLink } from './SearchLink';
+
+export const TableHead = () => {
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  function setSort(keySort: string) {
+    const sortParams = { sort, order };
+
+    if (sort !== keySort) {
+      sortParams.sort = keySort;
+      sortParams.order = null;
+    } else if (!order) {
+      sortParams.order = 'desc';
+    } else {
+      sortParams.sort = null;
+      sortParams.order = null;
+    }
+
+    return sortParams;
+  }
+
+  return (
+    <tr>
+      {['name', 'sex', 'born', 'died'].map(field => {
```

Comment:
Avoid magic strings. It would be better to define the array ['name', 'sex', 'born', 'died'] as a constant above the component.